src/lib/coerce.js

```diff
@@ -19,21 +19,70 @@ var nestedProperty = require('./nested_property');
 var counterRegex = require('./regex').counter;
 var modHalf = require('./mod').modHalf;
 var isArrayOrTypedArray = require('./array').isArrayOrTypedArray;
+var isTypedArraySpec = require('./array').isTypedArraySpec;
+var decodeTypedArraySpec = require('./array').decodeTypedArraySpec;
+var coerceTypedArraySpec = require('./array').coerceTypedArraySpec;
+
 
 exports.valObjectMeta = {
     data_array: {
         // You can use *dflt=[] to force said array to exist though.
         description: [
             'An {array} of data.',
-            'The value MUST be an {array}, or we ignore it.',
-            'Note that typed arrays (e.g. Float32Array) are supported.'
+            'The value could be an {array}',
+            'noting that typed arrays (e.g. Float32Array) are also supported.',
+            'It could also be an object in the form of',
+            'v: {, dtype: \'float32\', bvals: [/* ... */]}, shape: [dim0 (, dim1, (dim3))]',
+            'otherwise, it would be ignored.'
         ].join(' '),
         requiredOpts: [],
         otherOpts: ['dflt'],
         coerceFunction: function(v, propOut, dflt) {
-            // TODO maybe `v: {type: 'float32', vals: [/* ... */]}` also
-            if(isArrayOrTypedArray(v)) propOut.set(v);
-            else if(dflt !== undefined) propOut.set(dflt);
+            var wasSet;
+            if(isArrayOrTypedArray(v)) {
+                propOut.set(v);
+                wasSet = true;
+            } else if(isTypedArraySpec(v)) {
+                // Copy and coerce spec
+                v = coerceTypedArraySpec(v);
+
+                // See if caching location is available
+                var uid = propOut.obj.uid;
+                var module = propOut.obj._module;
+
+                if(v.bvals.constructor === ArrayBuffer || !uid || !module) {
+                    // Already an ArrayBuffer
+                    // decoding is cheap
+                    propOut.set(decodeTypedArraySpec(v));
+                } else {
+                    var prop = propOut.astr;
+                    var cache = module._b64BufferCache || {};
+
+                    // Check cache
+                    var cachedBuffer = ((cache[uid] || {})[prop] || {})[v.bvals];
```

Comment:
So `bvals` is a key (the only key) of the inner cache object? Do we know that this actually manages to use the same string in memory as the input object or does this end up duplicating the string? Might be safer to have the cache entry be a 2-item array `[v.bvals, decoded.buffer]` rather than an object `{v.bvals: decoded.buffer}`.

Reply:
Yeah, that's a great point. I don't know for certain whether JS would end up copying this. Doesn't seem like there'd be any reason to, but better to be safe here.

Reply:
Wondering if we could replace `bvals` key with `b64`?

Reply:
Let's not put the encoding into the sub-attribute name, in case we want to extend this to other encodings later. But I'm curious, does `bvals` come from somewhere else? If I was designing this in a vacuum I'd probably have called it just `vals` or `values` or something.

Reply:
Revised to use `vals` instead of `bvals`.